lib/diaspora/federation/entities.rb

```diff
@@ -0,0 +1,258 @@
+module Diaspora
+  module Federation
+    module Entities
+      def self.build(entity)
```

Comment:
Yes, @diaspora-code-review, you are perfectly right, the cyclomatic complexity is way too high for a normal function. But dude, take a look at the source! It's simply calling some function depending on the incoming objects, mate. That's not an issue! I am sure you can be happy with that, can't you?

Oh, wait, I am talking to a bot. It's going down.